code/parachain/runtime/dali/src/migrations.rs

```diff
@@ -19,6 +21,467 @@ impl OnRuntimeUpgrade for SchedulerMigrationV1toV4 {
 	}
 }
 
+pub mod hard_coded_assets {
+
+	use super::*;
+	use assets_registry::WeightInfo;
+	use composable_traits::{
+		assets::{AssetInfo, AssetInfoUpdate, InspectRegistryMetadata},
+		currency::Rational64,
+		rational,
+	};
+	use frame_support::WeakBoundedVec;
+
+	use primitives::topology;
+	use xcm::latest::prelude::*;
+
+	pub struct HardCodedAssetsMigration;
+
+	#[derive(Clone)]
+	struct AssetCreationInput {
+		asset_id: CurrencyId,
+		location: Option<XcmAssetLocation>,
+		asset_info: AssetInfo<Balance>,
+	}
+
+	impl AssetCreationInput {
+		fn new_asset(
+			asset_id: CurrencyId,
+			location: Option<XcmAssetLocation>,
+			name: Vec<u8>,
+			symbol: Vec<u8>,
+			decimals: u8,
+			existential_deposit: Balance,
+			ratio: Option<Rational64>,
+		) -> Self {
+			Self {
+				asset_id,
+				location,
+				asset_info: AssetInfo { name, symbol, decimals, existential_deposit, ratio },
+			}
+		}
+	}
+
+	fn asset_info_update(asset_info: AssetInfo<Balance>) -> AssetInfoUpdate<Balance> {
+		AssetInfoUpdate {
+			name: Some(asset_info.name),
+			symbol: Some(asset_info.symbol),
+			decimals: Some(asset_info.decimals),
+			existential_deposit: Some(asset_info.existential_deposit),
+			ratio: Some(asset_info.ratio),
+		}
+	}
+
+	fn add_assets_to_storage(assets: Vec<AssetCreationInput>) -> Weight {
+		let (mut count_created, mut count_updated) = (0, 0);
+		for asset_input in assets {
+			let AssetCreationInput { asset_id, location, asset_info } = asset_input;
+			// check if there is data stored for foreign asset
+			if let Some(foreign_location) = location.clone() {
+				// check that new asset_id is the same as old one for the same location
+				let mut location_stored = false;
+				if let Some(prev_asset_id) =
+					<AssetsRegistry as RemoteAssetRegistryInspect>::location_to_asset(
+						foreign_location.clone(),
+					) {
+					if prev_asset_id != asset_id {
+						panic!("previous and new asset_id for location do not match");
+					}
+					location_stored = true;
+				}
```

Comment:
NIT (can also be repeated below)
```suggestion
				let location_stored = <AssetsRegistry as RemoteAssetRegistryInspect>::location_to_asset(
						foreign_location.clone(),
					).map(|| {
					if prev_asset_id != asset_id {
						panic!("previous and new asset_id for location do not match");
					}
					true
					});
```